server/neptune/temporalworker/server.go

```diff
@@ -185,9 +195,17 @@ func (s Server) Start() error {
 	}()
 
 	// Start job output handler listener
-	// [WENGINES-4746] TODO: Clean up resources and exit gracefully on SIGTERM
 	go func() {
+		// Returns when the job output chan is closed which happens only after the temporal worker
+		// has gracefully shut down
 		s.JobStreamHandler.Handle()
+
+		// On cleanup, stream handler closes all active receivers and persists jobs in memory
+		ctx, cancel := context.WithTimeout(context.Background(), StreamHandlerTimeout)
+		defer cancel()
+		if err := s.JobStreamHandler.CleanUp(ctx); err != nil {
+			s.Logger.Error(err.Error())
+		}
```

Comment:
Why is this here? this function doesn't use a wg so its not guaranteed this will be called, also there might be a race with the stats/logger closing and this.

I would move this after the stop

Reply:
Ahh true!